File: swift/1. Two Sum.swift
class Solution {
  func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    var indexesByValue: [Int:Int] = [:]
    for (ix, x) in nums.enumerated() {
      if let iy = indexesByValue[target - x] {
        return [iy, ix]
      }
      indexesByValue[x] = ix
    }
    return []
  }
}

File: elixir/1. Two Sum.ex
defmodule Solution do
  @spec two_sum(nums :: [integer], target :: integer) :: [integer]
  def two_sum(nums, target) do
    nums
    |> Enum.reduce_while({0, %{}}, fn num, {index, indexes} ->
        delta = target - num
        first_index = indexes[delta]
        if nil != first_index do
            {:halt, [ first_index, index ]}
        else
            {:cont, {1+index, indexes |> Map.put(num, index)}}
        end
       end)
  end
end
